123-job-jwt/proposal.md

```diff
@@ -0,0 +1,137 @@
+# Summary
+
+Mint a signed JSON Web Token (JWT) and deliver to the job runtime container.
+Additionally, provide a validation endpoint to allow for external or internal
+validation of the issued JWT.
```

Comment:
Where do you still see usecases for internal validation? Personally, I think validation should happen on any system *consuming the JWT*, but I don't see a lot of need for Concourse itself to check it, except if some workloads use that JWT to talk to concourse itself (think about modifying the currently running pipeline…)

I think we should explicitly add a public `.well-known/openid-configuration`, that also contains the URL to a [JWKS](https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-key-sets) endpoint.

That's what other systems need to be able to validate JWTs, and what any concourse-internal services that workloads might need to talk to would also need to validate JWTs.

Reply:
Yes, I'll remove the `or internal` part of this sentence, I think the primary use case is external validation here, and if internal validation is required in the future, that can be a separate RFC.